src/components/DocumentControlsPanel/DocumentActionsPanel/DocumentActionsPanel.cypress.js

```diff
@@ -102,16 +102,16 @@ describe("DocumentActions", function() {
     cy.visit(`?collection=${this.collectionId}`);
 
     // Click delete to delete the only document
-    cy.get("[data-testid=delete_button]").click();
+    cy.get("[aria-label='Delete button']").click();
 
     // Click the X to close the Are you sure prompt
-    cy.get("[data-testid=modal_header]")
+    cy.get("[aria-label='Modal header']")
```

Comment:
Use `getByText` to access the header based on the text inside.

Reply:
This scopes me to the text, whereas the button we want to click is a uncle/aunt of the text. 

Would I basically do getByText then something like .parent().last() instead??

Reply:
You can search within a given DOM element. So first query for a relevant parent, (preferably using a visual hook that the user would use, such as a textual header). Then, make a query inside that element. More: https://testing-library.com/docs/dom-testing-library/api-helpers#within-and-getqueriesforelement-apis

Reply:
So in this particular instance, we are looking to use the "X" to close the modal. This "X" is actually rendered via Reactstrap.

There are no parent text hooks that would make sense (although, I think your overall point is still correct and relevant and helpful). See attachment for structure of the element.

In this case, because the "X" is technically an Icon and reactstrap is friendly enough to provide an aria-label of "close" already (and again, there is no parent text because of being a modal), it would make sense to click off that, correct?

so: 
`cy.get("[aria-label="close"]); `
and remove the children junk?

Reply:
Edit - the "x" is not an icon sorry, I was juggling a few projects for a second and misspoke there. But I think the point still remains because targeting the letter "x" by text seems really broad

Reply:
Oops, agreed. An aria-label makes sense for the x.